libraries/libstratosphere/source/settings/impl/settings_key_value_store.cpp

```diff
@@ -0,0 +1,1629 @@
+        R_UNLESS(++it != map->end(), ResultSettingsEndIteration());
+
+        /* Get the map key. */
+        const MapKey &map_key = it->first;
+        
+        /* Ensure the advanced iterator retains the required name. */
+        R_UNLESS(std::strncmp(map_key.GetString(), out->map_key, out->name_size) == 0, ResultSettingsEndIteration());
+
+        /* Ensure there is sufficient memory for the map key. */
+        const size_t map_key_size = map_key.GetCount() + 1;
+        R_UNLESS(GetHeapAllocatableSize() >= map_key_size, ResultSettingsKeyValueStoreKeyIteratorAllocationFailed());
+
+        /* Free the iterator's old map key. */
+        FreeToHeap(out->map_key, out->key_size);
+
+        /* Allocate the new map key. */
+        char *buffer = static_cast<char *>(AllocateFromHeap(map_key_size));
+        AMS_ASSERT(buffer != nullptr);
+
+        /* Copy the new map key to the buffer. */
+        std::memcpy(buffer, map_key.GetString(), map_key_size);
+
+        /* Set the output map key. */
+        out->key_size = map_key_size;
+        out->map_key = buffer;
+
+        return ResultSuccess();
+    }
+    
+    Result DestroyKeyValueStoreKeyIterator(KeyValueStoreKeyIterator *out) {
+        /* Check preconditions. */
+        AMS_ASSERT(out != nullptr);
+        AMS_ASSERT(out->name_size > 0);
+        AMS_ASSERT(out->name_size < out->key_size);
+        AMS_ASSERT(out->map_key != nullptr);
+
+        /* Acquire exclusive access to global state. */
+        std::scoped_lock lk(g_key_value_store_mutex);
+
+        /* Free the key to the heap. */
+        FreeToHeap(out->map_key, out->key_size);
+
+        /* Reset the name and key. */
+        out->name_size = 0;
+        out->key_size = 0;
+        out->map_key = nullptr;
+        return ResultSuccess();
+    }
+
+    Result GetKeyValueStoreItemCountForDebug(size_t *out_count) {
+        /* Check preconditions. */
+        AMS_ASSERT(out != nullptr);
+
+        /* Acquire exclusive access to global state. */
+        std::scoped_lock lk(g_key_value_store_mutex);
+
+        /* Get the key value store map. */
+        Map *map = nullptr;
+        R_TRY(GetKeyValueStoreMap(std::addressof(map)));
+        AMS_ASSERT(map != nullptr);
+
+        /* Output the item count. */
+        *out_count = map->size();
+        return ResultSuccess();
+    }
+    
+    Result GetKeyValueStoreItemForDebug(size_t *out_count, KeyValueStoreItemForDebug * const out_items, size_t out_items_count) {
+        /* Check preconditions. */
+        AMS_ASSERT(out_count != nullptr);
+        AMS_ASSERT(out_items != nullptr);
+
+        /* Acquire exclusive access to global state. */
+        std::scoped_lock lk(g_key_value_store_mutex);
+
+        /* Get the key value store map. */
+        Map *map = nullptr;
+        R_TRY(GetKeyValueStoreMap(std::addressof(map)));
+        AMS_ASSERT(map != nullptr);
+
+        /* Define the count variable. */
+        u64 count = 0;
```

Comment:
This can actually be size_t. Yes, it's confusing, since you'll need to change the actual function *out_count to be u64 *.